### ISSUE TITLE
Error: TypeError: Reduce of empty array with no initial value (Crunchyroll)

### ISSUE THREAD

**Issue by einkoro** [animeloyalty/animesync#28]
Fetching Puella Magi Madoka Magica (non-dubbed) seems to be consistently failing from Canada on Crunchyroll.

```
Fetching Puella Magi Madoka Magica 01 [Crunchyroll]
Error: TypeError: Reduce of empty array with no initial value
    at Array.reduce (<anonymous>)
    at Sync.saveAsync (/Users/einkoro/.nvm/versions/node/v14.15.2/lib/node_modules/animesync/dist/cli/classes/Sync.js:56:18)
    at async Object.downloadAsync (/Users/einkoro/.nvm/versions/node/v14.15.2/lib/node_modules/animesync/dist/cli/downloads/download.js:51:21)
    at async Object.seriesAsync (/Users/einkoro/.nvm/versions/node/v14.15.2/lib/node_modules/animesync/dist/cli/downloads/series.js:31:13)
    at async /Users/einkoro/.nvm/versions/node/v14.15.2/lib/node_modules/animesync/dist/cli/actions/download.js:33:17
    at async Function.usingAsync (/Users/einkoro/.nvm/versions/node/v14.15.2/lib/node_modules/animesync/dist/Server.js:42:16)
    at async Command.downloadAsync (/Users/einkoro/.nvm/versions/node/v14.15.2/lib/node_modules/animesync/dist/cli/actions/download.js:27:5)
Rejected https://www.crunchyroll.com/puella-magi-madoka-magica
```

I can provide the full log if needed.

**Comment by Deathspike** [comment]
Ah, that's a good one. The dub filter wasn't excluding the dub episodes as intended.
Fixed in https://github.com/animeloyalty/animesync/commit/1be3b4d0f19345271d14e83aa635b368ff7d6b3a and released as `0.11.5`.
Please let me know if this resolves your problem 👍

**Comment by einkoro** [comment]
I am still seeing it after updating to 0.11.5 – is there any stale data I might need to clear such as the episode tracking dot files?

**Comment by einkoro** [comment]
Looks like Kimi ni Todoke produces the same error as well on 0.11.5:
```
Fetching https://www.crunchyroll.com/kimi-ni-todoke-from-me-to-you
Fetching Kimi ni Todoke - From Me To You 01 [Crunchyroll]
Error: TypeError: Reduce of empty array with no initial value
    at Array.reduce (<anonymous>)
    at Sync.saveAsync (/Users/einkoro/.nvm/versions/node/v14.15.2/lib/node_modules/animesync/dist/cli/classes/Sync.js:56:18)
    at async Object.downloadAsync (/Users/einkoro/.nvm/versions/node/v14.15.2/lib/node_modules/animesync/dist/cli/downloads/download.js:51:21)
    at async Object.seriesAsync (/Users/einkoro/.nvm/versions/node/v14.15.2/lib/node_modules/animesync/dist/cli/downloads/series.js:31:13)
    at async /Users/einkoro/.nvm/versions/node/v14.15.2/lib/node_modules/animesync/dist/cli/actions/download.js:33:17
    at async Function.usingAsync (/Users/einkoro/.nvm/versions/node/v14.15.2/lib/node_modules/animesync/dist/Server.js:42:16)
    at async Command.downloadAsync (/Users/einkoro/.nvm/versions/node/v14.15.2/lib/node_modules/animesync/dist/cli/actions/download.js:27:5)
Rejected https://www.crunchyroll.com/kimi-ni-todoke-from-me-to-you
```

**Comment by Deathspike** [comment]
> 
> 
> I am still seeing it after updating to 0.11.5 – is there any stale/cached data I might need to clear such as the episode tracking dot files?

Nah. I missed another problem because I neglected to do a end-to-end test after fixing one bug. And there were two. Oops.

**Comment by Deathspike** [comment]
Should be fixed, for real this time, in https://github.com/animeloyalty/animesync/commit/90de0822d0b7056778dec8d1d0e5f9131ad332a1
Released as `0.11.6`. Please let me know if it works for you.

**Comment by einkoro** [comment]
Looks like its working with all four series now. 👍
